libcipher/caesar.py

```diff
@@ -0,0 +1,108 @@
+__author__ = 'kskrishnasangeeth'
+
+
+class Error(Exception):
+    pass
+
+
+class WrongKey(Error):
+    """ class created for throwing Exception in the event of wrong key being supplied
+
+    Attributes:
+        msg (str): Explanation on why key is not allowed.
+        key (int): Key that is supplied.
+
+    Args:
+        msg (str): Explanation on why key is not allowed.
+        key (int): Key that is supplied.
+
+   """
+    def __init__(self, msg, key):
+        self.msg = msg
+        self.key = key
+
+
+class NoKeyGiven(Error):
+    """ class created for throwing Exception in the event of no key being supplied
+
+    Attributes:
+        msg (str): Message to provide key for encryption.
+
+    Args:
+        msg (str): Message to provide key for encryption.
+
+    """
+    def __init__(self, msg):
+        self.msg = msg
+
+
+def cipher_helper(message, key):
+    """Function for modularising and removing some of the repeated code.
+
+    Args:
+        message (str): Message that is supplied for encryption/decryption.
+        key (int): Key used for encryption/decryption.
+
+    Returns:
+        message (str): Message in lowercase.
+
+    """
+    if key is None:
+        raise NoKeyGiven("Key is needed for Encryption.")
+
+    if key <= 0 or key >= 27:
+        raise WrongKey("Key Value should be between 0 and 26. Got Key as", key)
+
+    return message.lower()
+
+
+def encrypt(message, key=None):
```

Comment:
Why would you want to make it optional and later throw an error that user did not provide it?